x-pack/elastic-agent/docs/elastic-agent-command-line.asciidoc

```diff
@@ -8,40 +10,276 @@
+* <<elastic-agent-inspect-command,inspect>>
+* <<elastic-agent-run-command,run>>
+* <<elastic-agent-version-command,version>>
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-enroll-command]]
+== elastic-agent enroll
+
+Enroll the {elastic-agent} in {fleet}.
+
+We recommend that you run this command as the root user because some
+integrations require root privileges to collect sensitive data. This command
+overwrites the `elastic-agent.yml` file in the agent directory.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent enroll <kibana_url> <enrollment_token> [--ca-sha256 <string>]
+                     [--certificate-authorities <string>] [--force] [--help]
+                     [--insecure][global-flags]
+----
+
+[discrete]
+=== Options
+
+`kibana_url`::
+Required. URL of the {kib} endpoint where {fleet} is running.
+
+`enrollment_token`::
+Required. Enrollment token generated by {fleet}. You can use the same
+enrollment token for multiple agents.
+
+`--ca-sha256 <string>`::
+Comma-separated list of certificate authority hash pins used for certificate
+verification.
+
+`--certificate-authorities <string>`::
+Comma-separated list of root certificates used for server verification.
+
+`--force`::
+Force overwrite of current configuration without prompting for confirmation.
+This flag is helpful when using automation software or scripted deployments.
+
+`--help`::
+Show help for the `enroll` command.
+
+`--insecure`::
+Allow an insecure connection to {kib}. We strongly recommend that you use a
+secure connection.
+
+{global-flags-link}
+
+[discrete]
+=== Example
+
+[source,shell]
+----
+elastic-agent enroll http://localhost:5601 ZnmNIdzVITUJua2QIdU5FTWROVjY6dHY2N1EybWNTMUdPejg5ODbYcVpNUQ==
+----
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-help-command]]
+== elastic-agent help
+
+Show help for a specific command.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent help <command> [--help] [global-flags]
+----
+
+[discrete]
+=== Options
+
+`command`::
+The name of the command.
+
+`--help`::
+Show help for the `help` command.
+
+{global-flags-link}
+
+[discrete]
+=== Example
+
+[source,shell]
+----
+elastic-agent help enroll
+----
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-inspect-command]]
+== elastic-agent inspect
+
+Show the current {agent} configuration.
+
+If no parameters are specified, shows the full {agent} configuration.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent inspect [--help] [global-flags]
+elastic-agent inspect output [--output <string>] [--program <string>]
+                             [--help] [global-flags]
+----
+
+
+//REVIEWERS: The language used to describe this option is a little confusing. 
+//Beats users are likely to think of "output" as the output section for sending
+//to Elasticsearch or Logstash. Any suggestions for making this more clear?
+
+[discrete]
+=== Options
+
+`output`:: Display the configuration generated for the output. This command
```

Comment:
I assume the `generated` here comes from the use case with Ingest Manager. But in the end, it is just the current running output config. So perhaps remove generated and use `current configuration` instead?